### PR TITLE
feat: debugger guide

### PR DESCRIPTION
**Pull request by officialasishkumar**
## Related Issue
  Added debugger guide

Closes: #1002 

#### Describe the changes you've made
Added Debugger guide

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by officialasishkumar** [comment]
@shivamsouravjha Please take a look

File: debugger-guide.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+
+# Debugger Guide 👨‍🔧
+
+Here we will learn how to use debugger for Keploy in VS Code.
+
+To bring up the **Run and Debug** view, select the **Run and Debug** icon in the **Activity Bar** on the side of VS Code. You can also use the keyboard shortcut `Cmd + Shift + D` for MacOS and `Ctrl + shift + D` for Windows.
+
+If running and debugging is not yet configured (no `launch.json` has been created), VS Code shows the Run start view.
```
Comment (PranshuSrivastava) [inline debugger-guide.md:8]: Add screenshots as well to make it more descriptive

File: debugger-guide.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+```
+
+Let's take a closer look at some important key-value pairs in our JSON file:
+
+- The `"name"`  parameter can be anything, but for convenience, consider using the keploy command name (e.g., `Record` and `Test`).
+- The `args` parameter represents the entire keploy command to run your application. For example, in [Gin + Mongo](https://keploy.io/docs/quickstart/samples-gin/), on Linux, it's `record -c "go run main.go handler.go"`, and for MacOS, it's `record -c "docker compose up" --containerName "ginMongoApp"`.
+
+You can either add more objects in the "configurations" array or modify the "args" property within a single object to add more commands.
```
Comment (PranshuSrivastava) [inline debugger-guide.md:56]: Give an example here on how to use more flags with this configuration.

File: debugger-guide.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            "type": "go",
+            "request": "launch",
+            "mode": "auto",
+            "asRoot": true,
+            "console": "integratedTerminal",
+            "program": "main.go",
+            "args": [
+                "test",
+                "-c",
+                "<path_to_executable>",
```
Comment (PranshuSrivastava) [inline debugger-guide.md:43]: I think we can also use the docker run command here and not just the path to the executable file

**Review by PranshuSrivastava — CHANGES_REQUESTED** [review]

**Comment by officialasishkumar** [comment]
Done @PranshuSrivastava

File: debugger-guide.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    "type": "go",
+    "request": "launch",
+    "mode": "auto",
+    "asRoot": true,
+    "console": "integratedTerminal",
+    "program": "main.go",
+    "args": [
+        "test",
+        "-c",
+        "/home/coding/open-source/keploy/samples-go/gin-mongo/test-app-url-shortener",
```
Comment (PranshuSrivastava) [inline debugger-guide.md:72]: You haven't shown how to pass env variables, we also need that sometimes

File: debugger-guide.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
```
Comment (PranshuSrivastava) [inline debugger-guide.md:86]: don't use hardcoded values like this in examples, just say, path to binary

File: debugger-guide.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            "type": "go",
+            "request": "launch",
+            "mode": "auto",
+            "asRoot": true,
+            "console": "integratedTerminal",
+            "program": "main.go",
+            "args": [
+                "test",
+                "-c",
+                "<path_to_executable>", // We can also put docker run command here.
```
Comment (PranshuSrivastava) [inline debugger-guide.md:45]: there is no point of using a docker run command here, as the breakpoints will not work. Have you tested it before adding the comment?

**Review by PranshuSrivastava — CHANGES_REQUESTED** [review]

**Comment by PranshuSrivastava** [comment]
Closing as this has already been added.

**Pull request closed without merging**